### PR TITLE
Shops map

### LINKED ISSUE #41: Widok pobliskich sklepów, w których można zrealizować zakupy z listy

### PR DESCRIPTION
**Pull request by lychanl**
Closes #41

### CODE REVIEW

File: shoppingm8_fe/lib/list/product/productCategory.dart [hunk 1]
```diff
@@ -93,4 +93,24 @@ class ProductCategoryHepler {
...
+      case ProductCategory.CLEANING_SUPPLIES:
+        return { "supermarket" };
+      case ProductCategory.PHARMACY:
+        return { "pharmacy", "drugstore" };
+      case ProductCategory.COSMETICS:
+        return { "supermarket", "drugstore" };
+      case ProductCategory.TOOLS:
+        return { "hardware_store" };
+      case ProductCategory.HOBBY_OR_FUN:
+        return { "bicycle store", "book store", "pet store", "electronics_store", "florist", "jewelry_store" };
```
Comment (Delebrith) [inline shoppingm8_fe/lib/list/product/productCategory.dart]: just to be sure, is the lack of underscore intentional and does it work properly?

Comment (lychanl) [reply shoppingm8_fe/lib/list/product/productCategory.dart]: Fixed

File: shoppingm8_fe/lib/maps/placesProvider.dart [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+    loc.LocationData locationData = await location.getLocation();
+    return LatLng(locationData.latitude, locationData.longitude);
+  }
+
+  Future<Set<Marker>> _getPlacesByType(String type, LatLng location, int radius) async {
+    PlacesSearchResponse response =
+        await _connector.searchNearbyWithRadius(new Location(location.latitude, location.longitude), radius, type: type);
+    if (!response.isOkay) {
+      print("Error retrieving shops: " + (response.errorMessage ?? "null"));
```
Comment (Delebrith) [inline shoppingm8_fe/lib/maps/placesProvider.dart]: maybe we should display an error message to the user besides logging it

File: shoppingm8_fe/lib/maps/placesProvider.dart [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  }
+
+  Future<Set<Marker>> getPlacesByCategory(Set<ProductCategory> categories, location, {radius = 1000}) async {
+    Set<String> types = {};
+    for (ProductCategory category in categories) {
+      types.addAll(ProductCategoryHepler.getGooglePlaceType(category));
+    }
+
+    if (location == null) {
+      print("Failed to retrieve location");
```
Comment (Delebrith) [inline shoppingm8_fe/lib/maps/placesProvider.dart]: the same thing

**Review by Delebrith — CHANGES_REQUESTED** [review]
I think a toast with short error message will be sufficient, but we need somehow tell the user if something goes wrong

**Review by Delebrith — APPROVED** [review]

**Pull request merged**